app/controllers/CyaSupplementaryUnitController.scala

```diff
@@ -91,11 +92,16 @@ class CyaSupplementaryUnitController @Inject() (
           val finalHasSuppUnitOpt = model.hasSupplementaryUnit
           val finalSuppUnitOpt    = model.supplementaryUnit
 
-          val isValueChanged    =
+          val isValueChanged =
             initialHasSuppUnitOpt != finalHasSuppUnitOpt ||
               compareSupplementaryUnits(initialSuppUnitOpt, finalSuppUnitOpt)
+
+          val finalSuppUnitBD = convertToBigDecimal(finalSuppUnitOpt)
+
+          //if supplementary unit is zero, we consider it removed. This needs to be re-factored when API starts supporting null for removing objects
           val isSuppUnitRemoved =
-            initialHasSuppUnitOpt.contains(true) && finalHasSuppUnitOpt.contains(false)
+            (initialHasSuppUnitOpt
+              .contains(true) && finalHasSuppUnitOpt.contains(false)) || finalSuppUnitBD.contains(0)
```

Comment:
think we can do finalSuppUnitBD == 0 instead of contains as its already BigDecimal?

Reply:
Makes sense but can't compare BigDecimal with Int here, we should either have to use compareTo or equals as it is ab object so keeping it like this would be more straightforward I believe. happy to change if you feel we could use compareTo/equals